pkg/vendors/mellanox/mellanox.go

```diff
@@ -162,6 +163,14 @@ func (m *mellanoxHelper) MlxResetFW(pciAddresses []string) error {
 func (m *mellanoxHelper) MlxConfigFW(attributesToChange map[string]MlxNic) error {
 	log.Log.Info("mellanox-plugin configFW()")
 	for pciAddr, fwArgs := range attributesToChange {
+		bfMode, err := m.GetMellanoxBlueFieldMode(pciAddr)
+		if err != nil {
+			// NIC is not a DPU or mstconfig failed. It's safe to continue FW configuration
+			log.Log.V(2).Info("mellanox-plugin: configFW(): can't get DPU mode for NIC", "pciAddress", pciAddr)
+		}
+		if bfMode == BluefieldDpu {
+			return errors.Errorf("NIC %s is in DPU mode", pciAddr)
```

Comment:
can you elaborate that firmware configuration changes are not supported in this case ?
also add a comment saying why (because reboot will not re-load firmware)

Reply:
we need to document that carefully as we have the fwreset feature we implement in the operator but that is not to switch the BF card mode

Reply:
done